abb_hardware_interface/include/abb_hardware_interface/rws_client.hpp

```diff
@@ -0,0 +1,69 @@
+/***********************************************************************************************************************
+ *
+ * Copyright (c) 2020, ABB Schweiz AG
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with
+ * or without modification, are permitted provided that
+ * the following conditions are met:
+ *
+ *    * Redistributions of source code must retain the
+ *      above copyright notice, this list of conditions
+ *      and the following disclaimer.
+ *    * Redistributions in binary form must reproduce the
+ *      above copyright notice, this list of conditions
+ *      and the following disclaimer in the documentation
+ *      and/or other materials provided with the
+ *      distribution.
+ *    * Neither the name of ABB nor the names of its
+ *      contributors may be used to endorse or promote
+ *      products derived from this software without
+ *      specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ ***********************************************************************************************************************
+ */
+
+// This file is a modified copy from
+// https://github.com/ros-industrial/abb_robot_driver/tree/master/abb_rws_service_provider/include/abb_rws_service_provider/
+// https://github.com/ros-industrial/abb_robot_driver/tree/master/abb_rws_state_publisher/include/abb_rws_state_publisher/
+
+#ifndef ABB_HARDWARE_INTERFACE__RWS_CLIENT_HPP_
+#define ABB_HARDWARE_INTERFACE__RWS_CLIENT_HPP_
+
+// ROS
+#include <rclcpp/rclcpp.hpp>
+
+// ABB
+#include <abb_egm_rws_managers/rws_manager.h>
+#include <abb_egm_rws_managers/system_data_parser.h>
+
+namespace abb_rws_client {
+
+class RWSClient {
+ public:
+  RWSClient(const rclcpp::Node::SharedPtr &node, const std::string &robot_ip, unsigned short robot_port);
+
+ protected:
+  rclcpp::Node::SharedPtr node_;
+  abb::robot::RWSManager rws_manager_;
```

Comment:
Could you include the original comments from abb_ros_driver to this file documenting the variables?

Reply:
This also applies to all header files other than `mapping.hpp` and `utilities.hpp`